azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/JarArtifactHandlerImpl.java

```diff
@@ -0,0 +1,117 @@
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.plugin.MojoExecutionException;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Paths;
+
+/**
+ * Artifact handler for deploying a JAR, self-contained, Java application (e.g.
+ * Spring Boot) to Azure App Service through FTP
+ *
+ * @since 1.2.0
+ */
+public final class JarArtifactHandlerImpl extends FTPArtifactHandlerImpl {
+
+    public static final String FILE_IS_NOT_JAR = "The deployment file is not a jar typed file.";
+    public static final String FIND_JAR_FILE_FAIL = "Failed to find the jar file: '%s'";
+
+
+    public static final String DEFAULT_LINUX_JAR_NAME = "app.jar";
+    public static final String JAR_CMD = "__JAR_COMMAND__";
+    public static final String DEFAULT_JAR_COMMAND = "-Djava.net.preferIPv4Stack=true "
+            + "-Dserver.port=%HTTP_PLATFORM_PORT% "
+            + "-jar &quot;%HOME%\\\\site\\\\wwwroot\\\\FILENAME&quot;";
+    public static final String GENERATE_WEB_CONFIG_FAIL = "Failed to generate web.config file for JAR deployment.";
+    public static final String READ_WEB_CONFIG_TEMPLATE_FAIL = "Failed to read the content of web.config.template.";
+    public static final String GENERATING_WEB_CONFIG = "Generating web.config for Web App on Windows.";
+
+    public JarArtifactHandlerImpl(final AbstractWebAppMojo mojo) {
+        super(mojo);
+    }
+
+    @Override
+    public void publish() throws Exception {
+        final File jar = getJarFile();
+        assureJarFileExisted(jar);
+
+        prepareDeploymentFiles(jar);
+
+        uploadDirectoryToFTP();
+    }
+
+    private void prepareDeploymentFiles(File jar) throws IOException {
+        final File parent = new File(mojo.getDeploymentStageDirectory());
+        parent.mkdirs();
+
+        if (StringUtils.isNotEmpty(mojo.getLinuxRuntime())) {
+            Files.copy(jar, new File(parent, DEFAULT_LINUX_JAR_NAME));
+        } else {
+            Files.copy(jar, new File(parent, jar.getName()));
+            generateWebConfigFile(jar.getName());
+        }
+    }
+
+    private String getJarCommand() {
+        if (StringUtils.isEmpty(mojo.getJarCommand())) {
+            return DEFAULT_JAR_COMMAND;
+        }
+        return mojo.getJarCommand();
+    }
+
+    private void generateWebConfigFile(String jarFileName) throws IOException {
+        mojo.getLog().info(GENERATING_WEB_CONFIG);
+        final String templateContent;
+        try (final InputStream is = getClass().getResourceAsStream("web.config.template")) {
+            templateContent = IOUtils.toString(is, "UTF-8");
+        } catch (IOException e) {
+            mojo.getLog().error(READ_WEB_CONFIG_TEMPLATE_FAIL);
+            throw e;
+        }
+
+        final String webConfigFile = templateContent
+                .replaceAll(JAR_CMD, getJarCommand())
+                .replaceAll("FILENAME", jarFileName);
```

Comment:
`FILENAME`

Reply:
Good call. FILENAME should only be replaced when using default command.

Reply:
Addressed